executor/cook/io_helper.py

```diff
@@ -91,31 +91,49 @@ def process_output(label, out_file, max_bytes_read_per_line, out_fn, flush_fn):
         The string to associate in outputs
     out_file: a file object
         Provides that output from the child process
-    max_bytes_read_per_line: int
-        The maximum number of bytes to read per call to readline().
     out_fn: function(string)
         Function to output a string
     flush_fn: function()
         Function that flushes the output.
+    flush_interval_secs: number
+        The number of seconds to wait between flushing buffered data
+    max_bytes_read_per_line: int
+        The maximum number of bytes to read per call to readline().
 
     Returns
     -------
     Nothing.
     """
     logging.info('Starting to pipe {}'.format(label))
+
+    io_lock = Lock()
+    lines_buffered_event = Event()
+
+    def trigger_flush():
+        if lines_buffered_event.is_set():
+            with io_lock:
+                logging.info('Flushing contents {}'.format(label))
+                flush_fn()
```

Comment:
Is it possible for `flush_fn` to throw an error? If so we should handle it so the thread doesn't die.